lib/model/query/forms.js

```diff
@@ -159,12 +161,14 @@ createVersion.audit.withResult = true;
 
 // TODO: we need to make more explicit what .def actually represents throughout.
 // for now, enforce an extra check here just in case.
-const publish = (form) => ({ Forms }) => {
+const publish = (form) => ({ Forms, Datasets }) => {
   if (form.draftDefId !== form.def.id) throw Problem.internal.unknown();
 
+  const publishedAt = (new Date()).toISOString();
   return Promise.all([
     Forms._update(form, { currentDefId: form.draftDefId, draftDefId: null }),
-    Forms._updateDef(form, { draftToken: null, enketoId: null, publishedAt: (new Date()).toISOString() })
+    Forms._updateDef(form, { draftToken: null, enketoId: null, publishedAt }),
+    Datasets.publish(form.def.id, publishedAt)
```

Comment:
A question that might be better for an interactive review: this will be called for every form when it gets published so I just wanted to make sure this wouldn't have any negative effects if called on a form without any datasets. Maybe it could have a more non-committal name like "Datasets.publishIfPresent", "maybePublish", "possiblyPublish"... or could we check `entityRelated` on the form and only conditional run this?

Reply:
just realized, I have to add audit logs for publish as well

Reply:
You two may have discussed this already, but I think we'll also need to add this new audit action to Frontend.

Reply:
yes, I have added it in https://github.com/getodk/central-frontend/pull/641

Reply:
renamed publish function to publishIfExists